src/syscheckd/db/include/fimDBHelpersUTInterface.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Wazuh Syscheckd
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * September 23, 2021.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "FDBHMockInterface.hpp"
+#include "dbItem.hpp"
+
+#ifndef _FIMDB_HELPERS_UT_INTERFACE_
+#define _FIMDB_HELPERS_UT_INTERFACE_
+
+namespace FIMDBHelper
+{
+    template<typename T>
+#ifndef WIN32
+
+    void initDB(unsigned int sync_interval, unsigned int file_limit,
+                            fim_sync_callback_t sync_callback, logging_callback_t logCallback,
+                            std::shared_ptr<DBSync>handler_DBSync, std::shared_ptr<RemoteSync>handler_RSync)
+    {
+        FIMDBHelpersUTInterface::initDB(sync_interval, file_limit, sync_callback, logCallback, handler_DBSync, handler_RSync);
+    }
+#else
+
+    void initDB(unsigned int sync_interval, unsigned int file_limit, unsigned int registry_limit,
+                             fim_sync_callback_t sync_callback, logging_callback_t logCallback,
+                             std::shared_ptr<DBSync>handler_DBSync, std::shared_ptr<RemoteSync>handler_RSync)
+    {
+        FIMDBHelpersUTInterface::initDB(sync_interval, file_limit, registry_limit, sync_callback, logCallback, handler_DBSync,
+                              handler_RSync);
+    }
+#endif
+
+    template<typename T>
```

Comment:
What is the variable that is used like a template?

Reply:
Inside the function we do not make use of any template variable. The main reason why it's declared as a template is to avoid breaking the calls to the production version of FimDBHelpers. Those functions are templatized, so this way depending on the context (wazuh_unit_testing enabled) we can just add a different namespace file but with the same namespace. From the point of view of the wrappers is as it's the same namespace.